nixpkgs/repositories.bzl

```diff
@@ -27,3 +33,17 @@ def rules_nixpkgs_dependencies():
         url = "https://github.com/bazelbuild/rules_java/releases/download/4.0.0/rules_java-4.0.0.tar.gz",
         sha256 = "34b41ec683e67253043ab1a3d1e8b7c61e4e8edefbcad485381328c934d072fe",
     )
+    if not local:
+        maybe(
+            http_archive,
+            "rules_nixpkgs_core",
+            # XXX: no way to use `sha256` here, but if this surrounding repo comes
+            # from that URL, Bazel should hit the cache for the sub-workspace
```

Comment:
I'm not sure it will, AFAIK Bazel uses the sha256 as the cache key.

While looking into this I found the following: [`native.existing_rule`](https://docs.bazel.build/versions/main/skylark/lib/native.html#existing_rule) exposes the attribute of previously defined (repository) rules as well. I think we could construct something like the following (this is only a very rough sketch)
```
def rules_nixpkgs_dependencies(*, rules_nixpkgs_url = "", rules_nixpkgs_sha256 = ""):
    # *_url|sha256 attributes for fallback/user override
    if rules_nixpkgs_url:
        # Use user override ...
    rules_nixpkgs = native.existing_rule("io_tweag_rules_nixpkgs")
    if rules_nixpkgs:
        kind = rules_nixpkgs.get("kind")
        if kind == "local_repository":
            path = rules_nixpkgs.get("path")
            maybe(
                local_repository,
                "rules_nixpkgs_core",
                path = path + "/core",
            )
        elif kind == "http_archive":
            url = rules_nixpkgs.get("url")
            urls = rules_nixpkgs.get("urls")
            sha256 = rules_nixpkgs.get("sha256")
            strip_prefix = rules_nixpkgs.get("sha256")
            # ... other relevant attributes, e.g. patches?
            maybe(
                http_archive,
                "rules_nixpkgs_core",
                url = url,
                urls = urls,
                sha256 = sha256,
                strip_prefix = strip_prefix + "/core",
                ...
            )
        else:
            # Fall back to user provided override or fail if missing
```
(Again, only a very rough sketch and missing a couple edge-cases and error handling)

Reply:
Thanks for the hint. Implemented it in a reasonably general fashion.